packages/grafana-ui/src/components/Table/TableInputCSV.tsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import debounce from 'lodash/debounce';
+import Papa, { ParseError, ParseMeta } from 'papaparse';
+import { TableData, Column } from '../../types/data';
+
+// Subset of all parse options
+export interface ParseConfig {
+  delimiter?: string; // default: ","
+  newline?: string; // default: "\r\n"
+  quoteChar?: string; // default: '"'
+  encoding?: string; // default: ""
+  comments?: boolean | string; // default: false
+}
+
+export interface ParseResults {
+  table: TableData;
+  meta: ParseMeta;
+  errors: ParseError[];
+}
+
+// This mutates the table structure!
+export function checkAndFix(table: TableData): number {
+  let cols = table.columns.length;
+  let different = 0;
+  table.rows.forEach(row => {
+    if (cols !== row.length) {
+      different++;
+      cols = Math.max(cols, row.length);
+    }
+  });
+  if (different > 0) {
+    if (cols !== table.columns.length) {
+      const diff = cols - table.columns.length;
+      for (let i = 0; i < diff; i++) {
+        table.columns.push({
+          text: 'Column ' + table.columns.length,
+        });
+      }
+    }
+    table.rows.forEach(row => {
+      const diff = cols - row.length;
+      for (let i = 0; i < diff; i++) {
+        row.push(null);
+      }
+    });
+  }
+  return different;
```

Comment:
Since there is quite a lot going on in this function I would rather return a tuple `[fixedTable, requiredFix]` rather than mutate the input table and return number of differences (why this information would be useful?)

Reply:
good point.  I will rework it to be a simple function that just returns the fixed table.  If you want to check if fixes were required (it shows an indicator in the toolbar) you can just call `output === input`